src/ensembl/src/content/app/browser/track-panel/track-panel-list/TrackPanelListItem.test.tsx

```diff
@@ -26,92 +30,177 @@ import {
 
 import { createMainTrackInfo } from 'tests/fixtures/track-panel';
 import { Status } from 'src/shared/types/status';
+import { DrawerView } from 'src/content/app/browser/drawer/drawerState';
 
-describe('<TrackPanelListItem />', () => {
-  beforeEach(() => {
-    jest.resetAllMocks();
-  });
+import * as drawerActions from '../../drawer/drawerActions';
+import * as browserActions from 'src/content/app/browser/browserActions';
+import * as trackPanelActions from 'src/content/app/browser/track-panel/trackPanelActions';
+import { isTrackCollapsed } from 'src/content/app/browser/track-panel/trackPanelSelectors';
+
+jest.mock('../../drawer/drawerActions', () => {
+  const originalModule = jest.requireActual('../../drawer/drawerActions');
 
-  const defaultProps: TrackPanelListItemProps = {
-    categoryName: faker.lorem.words(),
-    trackStatus: Status.SELECTED,
-    defaultTrackStatus: Status.SELECTED,
-    track: createMainTrackInfo(),
-    activeGenomeId: faker.lorem.words(),
-    activeEnsObjectId: faker.lorem.words(),
-    isDrawerOpened: false,
-    drawerView: 'bookmarks',
-    highlightedTrackId: faker.lorem.words(),
-    isCollapsed: false,
+  return {
+    ...originalModule,
     changeDrawerView: jest.fn(),
-    toggleDrawer: jest.fn(),
-    updateTrackStatesAndSave: jest.fn(),
+    setActiveDrawerTrackId: jest.fn()
+  };
+});
+
+jest.mock('src/content/app/browser/browserActions', () => {
+  const originalModule = jest.requireActual(
+    'src/content/app/browser/browserActions'
+  );
+
+  return {
+    ...originalModule,
+    updateTrackStatesAndSave: jest.fn()
+  };
+});
+
+jest.mock('src/content/app/browser/track-panel/trackPanelActions', () => {
+  const originalModule = jest.requireActual(
+    'src/content/app/browser/track-panel/trackPanelActions'
+  );
+
+  return {
+    ...originalModule,
     updateCollapsedTrackIds: jest.fn()
   };
+});
+
+const fakeGenomeId = faker.lorem.words();
+
+const mockState = {
+  drawer: {
+    isDrawerOpened: { [fakeGenomeId]: true },
+    drawerView: { [fakeGenomeId]: DrawerView.BOOKMARKS },
+    activeDrawerTrackIds: {}
+  },
+  browser: {
+    browserEntity: {
+      activeGenomeId: fakeGenomeId,
+      activeEnsObjectIds: {
+        [fakeGenomeId]: faker.lorem.words()
+      }
+    },
+    trackPanel: {
+      [fakeGenomeId]: {
+        highlightedTrackId: faker.lorem.words(),
+        collapsedTrackIds: []
+      }
+    }
+  }
+};
+
+jest.mock('react-redux', () => {
+  const originalModule = jest.requireActual('react-redux');
+  return {
+    ...originalModule,
+    useDispatch: () => jest.fn(),
+    useSelector: (selector: any) => {
+      return selector(mockState);
+    }
+  };
+});
```

Comment:
We shouldn't need to mock react-redux. In fact, I have a vague suspicion that if you remove this mock, the tests will continue working just fine.

Reply:
I tried removing it but it doesn't seem to work.

Reply:
I updated the tests and pushed a new commit.

In your tests, you were trying to tests that action creators were called by mocking the action creators with functions that return nothing. Redux didn't like that — it wants to receive actions that are objects containing at least the type field. I changed the tests to use the real action creators, and to check which actions were generated by getting them from the mock store. I suppose, it would also be possible to make mock action creators return some mock action object, and that would also work.

Reply:
Great! Thanks. I spent quite a long time trying to find a workaround for that error that was getting thrown when the `type` field was missing and I stopped looking further once I got it to work. the changes you have made seems to make more sense now.